modules/examiner/php/NDB_Form_examiner.class.inc

```diff
@@ -54,16 +54,23 @@ class NDB_Form_Examiner extends NDB_Form
         }
 
         if ($useCertification) {
-            $centerID = $DB->pselectOne(
-                "SELECT centerID 
-                 FROM examiners
-                 WHERE examinerID=:EID",
+            $cids      = $DB->pselect(
+                "SELECT epr.centerID 
+                 FROM examiners e 
+                  LEFT JOIN examiners_psc_rel epr ON (e.examinerID=epr.examinerID)
+                 WHERE e.examinerID=:EID",
                 array('EID' => $this->identifier)
             );
+            $centerIDs =array();
+            foreach ($cids as $k=>$row) {
+                $centerIDs[] = $row['centerID'];
+            }
 
+            // Access is only granted if user ENGLOBES all sites of examiner
+            // considering that certification changes affect all sites
             return $user->hasPermission('examiner_view')
                 && (($user->hasPermission('examiner_multisite')
-                || in_array($centerID, $user->getData('CenterIDs'))));
+                || empty(array_diff($centerIDs, $user->getData('CenterIDs')))));
```

Comment:
This line is starting to get convoluted.. maybe add a `$user->hasAllCenterIDs(array())` (or `$user->englobesCenterIDs()`) function? It seems like something that will be used in multiple places and should live in the user class.